lib/lightning/impact_tracking/client.ex

```diff
@@ -6,11 +6,24 @@ defmodule Lightning.ImpactTracking.Client do
   """
   use Tesla, only: [:post], docs: false
 
+  alias Lightning.ImpactTracking.ResponseProcessor
+
   def submit_metrics(metrics, host) do
-    build_client(host) |> post("/api/metrics", metrics)
+    result =
+      build_client(host)
+      |> post("/api/metrics", metrics)
+      |> elem(1)
+      |> ResponseProcessor.successful?()
+      |> indicate_success
```

Comment:
or better we could use a `then` here to avoid the indirection like this:
|> ResponseProcessor.successful?()
|> then(fn 
  true -> :ok
  false -> :error
end)